### DISCUSSION TITLE
Are the OverlayProvider and OverlayContainer components still needed with useModalOverlay and usePopover?

### DISCUSSION THREAD

**Discussion by angusd3v** [adobe/react-spectrum#4582] [Q&A]
The documentation now no longer mentions either of these components. I am having trouble with a modal dialog's focus trap not working, and I'm not sure if it's because of something to do with a misconfiguration of those components, but I can no longer find any reference to them. 

Could anyone clarify?

**Comment by angusd3v** [comment]
My confusion comes from this example in the docs for `useModalOverlay`. `useModalOverlay` calls `useOverlayFocusContain`, but what context is that `useOverlayFocusContain` actually talking to? I don't understand how the `Overlay` component rendered inside here would ever get its `<FocusScope />` `contain` prop set to `true` unless I were to call `useModalOverlay` inside a _child component_ rendered inside the `Overlay` here?

```
function Modal({ state, children, ...props }) {
  let ref = React.useRef(null);
  let { modalProps, underlayProps } = useModalOverlay(props, state, ref);

  return (
    <Overlay>
      <div
        style={{
          position: 'fixed',
          zIndex: 100,
          top: 0,
          left: 0,
          bottom: 0,
          right: 0,
          background: 'rgba(0, 0, 0, 0.5)',
          display: 'flex',
          alignItems: 'center',
          justifyContent: 'center'
        }}
        {...underlayProps}
      >
        <div
          {...modalProps}
          ref={ref}
          style={{
            background: 'var(--page-background)',
            border: '1px solid gray'
          }}
        >
          {children}
        </div>
      </div>
    </Overlay>
  );
}
```

**Reply by LFDanLu** [reply]
In the `useModalOverlay` docs example, the `Modal` renders a [Dialog](https://react-spectrum.adobe.com/react-aria/useModalOverlay.html#dialog) within it, which calls `useDialog`. `useDialog` in turn calls `useOverlayFocusContain` [here](https://github.com/adobe/react-spectrum/blob/48d2b91a4e0971e4697b086569dad571f3588e65/packages/%40react-aria/dialog/src/useDialog.ts#L62), which then sets the Overlay's `<FocusScope />` `contain` prop set to `true`.

**Reply by LFDanLu** [reply]
And just to clarify the original question about "Are the OverlayProvider and OverlayContainer components still needed with useModalOverlay and usePopover": no, using `usePopover`, `useModalOverlay`, and `Overlay` should be sufficient.
